agent/grpc-external/services/peerstream/stream_resources.go

```diff
@@ -204,6 +205,25 @@ func (s *Server) HandleStream(streamReq HandleStreamRequest) error {
 	)
 	subCh := mgr.subscribe(streamReq.Stream.Context(), streamReq.LocalID, streamReq.PeerName, streamReq.Partition)
 
+	// We need a mutex to protect against simultaneous sends to the client.
+	var sendMutex sync.Mutex
+
+	// streamSend is a helper function that sends msg over the stream
```

Comment:
This helper wraps the mutex and also extracts `logTraceSend` and `status.TrackSendError`. You'll see those calls removed from where we use the helper now.